pkg/security/loaders/raiffeisench/raiffeisench.go

```diff
@@ -0,0 +1,156 @@
+package raiffeisench
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/enrichman/portfolio-perfomance/pkg/security"
+)
+
+type RaiffeisenchQuoteLoader struct {
+	name  string
+	isin  string
+	valor int
+}
+
+func New(name, isin string) *RaiffeisenchQuoteLoader {
+	valor, err := getValorFromISIN(isin)
+	if err != nil {
+		panic(err)
+	}
+
+	return &RaiffeisenchQuoteLoader{
+		name:  name,
+		isin:  isin,
+		valor: valor,
+	}
+}
+
+func getValorFromISIN(isin string) (int, error) {
+	// https://en.wikipedia.org/wiki/Valoren_number
+	// https://www.isin.org/isin-format/
+
+	if !strings.HasPrefix(isin, "CH") {
+		return 0, fmt.Errorf("valor number can only be extracted for Swiss financial instruments")
+	}
+
+	// strip CH
+	val := strings.TrimPrefix(isin, "CH")
+
+	// strip all leading zeros
+	val = strings.TrimLeft(val, "0")
+
+	// drop last digit (ISIN checksum)
+	val = val[0 : len(val)-1]
+
+	valor, err := strconv.Atoi(val)
+	if err != nil {
+		return 0, fmt.Errorf("failed to read valor number from ISIN: %w", err)
+	}
+	return valor, nil
+}
+
+func (r *RaiffeisenchQuoteLoader) Name() string {
+	return r.name
+}
+
+func (r *RaiffeisenchQuoteLoader) ISIN() string {
+	return r.isin
+}
+
+func (r *RaiffeisenchQuoteLoader) Valor() int {
+	return r.valor
+}
+
+type HistoryQuotesRequest struct {
+	Valor      int       `json:"valor"`
+	ExchangeId int       `json:"exchangeId"`
+	CurrencyId int       `json:"currencyId"`
+	From       time.Time `json:"from"`
+	To         time.Time `json:"to"`
+}
+
+type HistoryQuotesResponse struct {
+	HistoryQuotes []struct {
+		Date  string      `json:"date"`
+		Close float64     `json:"close"`
+		Open  interface{} `json:"open"`
+		High  interface{} `json:"high"`
+		Low   interface{} `json:"low"`
+	} `json:"historyQuotes"`
+}
+
+func (r *RaiffeisenchQuoteLoader) LoadQuotes() ([]security.Quote, error) {
+	// Build request payload
+	endDate := time.Now()
+	startDate := time.Now().Add(-1 * time.Hour * 24 * 365) // one year ago
+	payload := HistoryQuotesRequest{
+		Valor:      r.valor,
+		ExchangeId: 3233,
```

Comment:
I actually don't know. It shows up in the URL when you visit this page https://boerse.raiffeisen.ch/fonds/detail/56145861 and is also used in the API call.

Reply:
Ok, it looks like it's the ID of the exchange market. For example if you go to https://boerse.raiffeisen.ch/it-ch/fondi/dettaglio/56145861 withit will fallback to `exchangeid=393`, the `SFD`. But at the bottom you can select a differen exchange, for example the `EPF CHF`, with the [exchangeid=3233](https://boerse.raiffeisen.ch/it-ch/fondi/dettaglio/56145861?exchangeid=3233)

I would move this at the top in a const, like:

```go
const EPFCHFExchangeID = 3233;
```

This will probably fail for some securities but I will leave the right fix on another PR.